lib/maths/CTimeSeriesModel.cc

```diff
@@ -96,6 +98,51 @@ TCalculation2Vec expand(maths_t::EProbabilityCalculation calculation) {
     return {};
 }
 
+//! Update \p probability to account for empty buckets for sparse data.
+double featureProbabilityGivenBucket(TBool2Vec empty,
+                                     maths_t::EProbabilityCalculation calculation,
+                                     TDouble2Vec value,
+                                     TDouble2Vec probabilityBucketEmpty,
+                                     double probability) {
+
+    // The value is tested against zero because it is assumed that if a feature
+    // is non-null for an empty bucket it's value is defined to be zero. This is
+    // true for all our current features, i.e. counts and sums.
+
+    auto pOneSide = [&](double pBucketEmpty) {
```

Comment:
nit: perhaps just capture `probability` in this lambda (and the next) rather than using the default lambda capture mode?